### DISCUSSION TITLE
Any examples on calculating the co2 depressurization according to GERG-2008 EOS?

### DISCUSSION THREAD

**Discussion by huisky2000** [equinor/neqsim#243] [General]
Hi,

I'm wondering do you have any example as the title says or similar ones?

Regards
Huisky

**Comment by EvenSol** [comment]
Hi.
You can find an example of calculation of CO2 depressurization in [this notebook](https://colab.research.google.com/github/EvenSol/NeqSim-Colab/blob/master/notebooks/thermodynamics/ThermodynamicandphysicalpropertiesofCO2.ipynb) and in [this Java file](https://github.com/equinor/neqsim/blob/master/src/test/java/neqsim/thermodynamicOperations/util/example/SolidFlash.java).

However the example is not working with the GERG-2008 EoS (GERG-2008 EOS is only implemented to calculate gas properties at the moment). The GERG-2008 EoS is implemented in a simplified way, and this does not support flash calculations at the moment. In NeqSim you can calculate the depressurization process using an EoS such as SRK/PR/UMR etc.

Best regards,
Even

**Reply by huisky2000** [reply]
Hi,
Thanks a lot for your reply. Are you planning to implement flash calculations by using the GERG-2008 EOS? Will there be significant differences between SRK/PR/UMR etc. from GERG-2008?

Regards
Huisky

**Reply by EvenSol** [reply]
It would be very nice to implement the GERG-2008 EoS as standard thermodynamic model in NeqSim. Then all flashes can be used (and also process simulations). There are no plans at the moment, but it could be a nice work for a student.

Very often the accuracy of SRK/PR is sufficient for calculation of properties of CO2 (eg. for calculation of temperature drop in valves). But if very high accuracy is needed, reference equations will be better.

**Comment by EvenSol** [comment]
You can add the following code to the end of the notebook to get properties using GERG-2008.

'''
// Outputs:
// P - Pressure (kPa)
// Z - Compressibility factor
// dPdD - First derivative of pressure with respect to density at constant
// temperature [kPa/(mol/l)]
// d2PdD2 - Second derivative of pressure with respect to density at constant
// temperature [kPa/(mol/l)^2]
// d2PdTD - Second derivative of pressure with respect to temperature and
// density [kPa/(mol/l)/K]
// dPdT - First derivative of pressure with respect to temperature at constant
// density (kPa/K)
// U - Internal energy (J/mol)
// H - Enthalpy (J/mol)
// S - Entropy [J/(mol-K)]
// Cv - Isochoric heat capacity [J/(mol-K)]
// Cp - Isobaric heat capacity [J/(mol-K)]
// W - Speed of sound (m/s)
// G - Gibbs energy (J/mol)
// JT - Joule-Thomson coefficient (K/kPa)
// Kappa - Isentropic Exponent
// A - Helmholtz energy (J/mol)
'''
print(fluid1.getPhase('gas').getProperties_GERG2008())
